dashpay/src/main/kotlin/org/dashj/platform/dashpay/BlockchainIdentity.kt

```diff
@@ -1841,4 +1886,52 @@ class BlockchainIdentity {
         val wif = privateKey?.getPrivateKeyEncoded(wallet!!.params)
         return "assetlocktx=$txid&pk=$wif&du=${currentUsername!!}&islock=${cftx.confidence.instantSendlock.toStringHex()}"
     }
+
+    // Transaction Metadata Methods
+    @Throws(KeyCrypterException::class)
+    fun publishTxMetaData(txMetadataItems: List<TxMetadataItem>, keyParameter: KeyParameter?) {
+        val keyIndex = 1
+        val encryptionKeyIndex = 0
+        val encryptionKey = privateKeyAtPath(keyIndex, TxMetadataDocument.childNumber, encryptionKeyIndex, IdentityPublicKey.Type.ECDSA_SECP256K1, null)
```

Comment:
Currently, these indexes are hard coded, but the `encryptionKeyIndex` should be incremented each time we publish tx metadata (so a new key is used to encrypted the metadata).

However, we don't yet have a way to store the last used value, especially after `this` is destroyed and recreated.  Some information in this class is stored in the Room DB DashPay.

We can figure this out in the next story where we integrate this feature into DashPay.